boudicca.base/publisher-event-html/src/test/kotlin/base/boudicca/publisher/event/html/PublisherHtmlApplicationE2ETests.kt

```diff
@@ -0,0 +1,307 @@
+    whenever(searchServiceCaller!!.getFiltersFor(any())).thenReturn(filters)
+
+    page.navigate("http://localhost:$port")
+
+    page.getByTestId("search-field").fill("Cultural")
+    page.getByTestId("search-field").press("Enter")
+
+    val events = page.querySelectorAll(".event")
+    val eventSize = events.size
+
+    assertTrue(eventSize == 1) { "Expected 1 event, but found $eventSize events." }
+    assertThat(page.locator(".event")).containsText("Cultural")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(ListOfFilterableEvents::class)
+  fun shouldBeAbleToFilterEvents(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    whenever(searchServiceCaller!!.search(any())).then {
+      val queryArgument = it.arguments.first() as QueryDTO
+      if (queryArgument.query.contains("Linz")) {
+        SearchResultDTO(
+          listOf(
+            Event("Cultural Event at Posthof", OffsetDateTime.now(), mapOf("city" to "Linz"))
+          ),
+          1,
+          null
+        )
+      } else {
+        SearchResultDTO(
+          events,
+          events.size,
+          null
+        )
+      }
+    }
+    whenever(searchServiceCaller!!.getFiltersFor(any())).thenReturn(filters)
+
+    page.navigate("http://localhost:$port")
+
+    page.locator("button[id=\"filterButton\"]").click()
+
+    // It is important to waitFor() the page to be in the desired
+    // state *before* running analyze(). Otherwise, axe might not
+    // find all the elements your test expects it to scan.
+    page.locator("#drawer").waitFor()
+
+    page.locator("[name='locationCity']").selectOption("Linz")
+    page.locator("button[id='filterSearchButton']").click()
+
+    page.waitForFunction(
+      "() => !document.querySelector('#drawer').classList.contains('drawer-open')"
+    )
+
+    val events = page.querySelectorAll(".event")
+    val eventSize = events.size
+
+    assertTrue(eventSize == 1) { "Expected 1 event, but found $eventSize events." }
+    assertThat(page.locator(".event")).containsText("Cultural")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(ListOfFilterableEvents::class)
+  fun shouldBeAbleToResetFilteredEvents(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    whenever(searchServiceCaller!!.search(any())).then {
+      val queryArgument = it.arguments.first() as QueryDTO
+      if (queryArgument.query.contains("Linz")) {
+        SearchResultDTO(
+          listOf(
+            Event("Cultural Event at Posthof", OffsetDateTime.now(), mapOf("city" to "Linz"))
+          ),
+          1,
+          null
+        )
+      } else {
+        SearchResultDTO(
+          events,
+          events.size,
+          null
+        )
+      }
+    }
+    whenever(searchServiceCaller!!.getFiltersFor(any())).thenReturn(filters)
+
+    page.navigate("http://localhost:$port")
+
+    page.locator("button[id=\"filterButton\"]").click()
+
+    // It is important to waitFor() the page to be in the desired
```

Comment:
you don't have to repeat this comment every time

Reply:
done